server/datastore/mysql/apple_mdm.go

```diff
@@ -1164,3 +1164,81 @@ func (ds *Datastore) InsertMDMIdPAccount(ctx context.Context, account *fleet.MDM
 	_, err := ds.writer.ExecContext(ctx, stmt, account.UUID, account.Username, account.Salt, account.Entropy, account.Iterations)
 	return ctxerr.Wrap(ctx, err, "creating new MDM IdP account")
 }
+
+func (ds *Datastore) GetMDMAppleFileVaultSummary(ctx context.Context, teamID *uint) (*fleet.MDMAppleFileVaultSummary, error) {
+	sqlFmt := `
+	SELECT
+	COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND hdek.decryptable = 1
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'applied'
+				AND hmap.operation_type = 'install') THEN
+			1
+		END) AS applied, COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND(hdek.decryptable = 0
+					OR hdek.decryptable IS NULL)
```

Comment:
This is the case where the key is either non-decryptable or we did not get the key yet, so it would require making sure the `hdek` row does not exist, something like `OR (hdek.host_id IS NULL AND hdek-decryptable IS NULL)` (basing those conditions on the corresponding implementation here: https://github.com/fleetdm/fleet/blob/main/server/fleet/hosts.go#L316-L379)